packages/dds/tree/src/feature-libraries/schemaIndex.ts

```diff
@@ -15,13 +15,169 @@ import { SummaryTreeBuilder } from "@fluidframework/runtime-utils";
 import { Index, SummaryElement, SummaryElementParser, SummaryElementStringifier } from "../shared-tree-core";
 import { cachedValue, ICachedValue, recordDependency } from "../dependency-tracking";
 import { Delta } from "../tree";
-import { StoredSchemaRepository } from "../schema-stored";
+import {
+    FieldKindIdentifier, FieldSchema, GlobalFieldKey, LocalFieldKey, Named,
+    SchemaData, StoredSchemaRepository, TreeSchema, TreeSchemaIdentifier, ValueSchema,
+} from "../schema-stored";
+import { brand } from "../util";
 
 /**
  * The storage key for the blob in the summary containing schema data
  */
 const schemaBlobKey = "SchemaBlob";
 
+const schemaStringKey = "SchemaString";
+
+const version = "1.0.0" as const;
+
+/**
+ * Format for encoding as json.
+ *
+ * For consistency all lists are sorted and undefined values are omitted.
+ *
+ * This chooses to use lists of named objects instead of maps:
+ * this choice is somewhat arbitrary, but avoids user data being used as object keys,
+ * which can sometimes be an issue (for example handling that for __proto__ can require care).
+ */
+interface Format {
+    version: typeof version;
+    tree: TreeSchemaFormat[];
+    field: NamedFieldSchemaFormat[];
```

Comment:
nit for consistency:
```suggestion
    trees: TreeSchemaFormat[];
    fields: NamedFieldSchemaFormat[];
```